### PR TITLE
put command discovery before scripts for Unix

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Also cleaned up the NativeLinuxCommands.Tests.ps1 script on skipping tests

Fix https://github.com/PowerShell/PowerShell/issues/2095

### CODE REVIEW

File: test/powershell/Language/Scripting/NativeExecution/NativeLinuxCommands.Tests.ps1 [hunk 1]
```diff
@@ -1,39 +1,57 @@
...
+        $originalPath = $env:PATH
+        $env:PATH += [IO.Path]::PathSeparator + $TestDrive
+    }
+
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParams
+        $env:PATH = $originalPath
+    }
+
     It "Should return a type of 'string' for hostname cmdlet" {
```
Comment (iSazonov) [inline test/powershell/Language/Scripting/NativeExecution/NativeLinuxCommands.Tests.ps1]: Now we skip the test - why?

**Review by iSazonov — APPROVED** [review]
LGTM with one minor comment.

File: test/powershell/Language/Scripting/NativeExecution/NativeLinuxCommands.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply test/powershell/Language/Scripting/NativeExecution/NativeLinuxCommands.Tests.ps1]: This test was created a long time ago.  I think it's actually not useful and is implicitly covered by tests that follow.  I'll remove it.

Comment (iSazonov) [reply test/powershell/Language/Scripting/NativeExecution/NativeLinuxCommands.Tests.ps1]: Closed.

**Review by anmenaga — APPROVED** [review]
LGTM
Why adding testdrive to $env:PATH is necessary?

**Comment by SteveL-MSFT** [comment]
@anmenaga I was thinking that would be a good way for command discovery to find the "native command" and the powershell script to validate the command was executed over the script without polluting the current working directory.

**Comment by SteveL-MSFT** [comment]
@mirichmo can you merge?

**Comment by daxian-dbw** [comment]
Is this change only for Unix? The title seems so but the change is not guarded by `#if UNIX`.

**Review by mirichmo — APPROVED** [review]

**Comment by mirichmo** [comment]
@daxian-dbw The code is guarded  by a runtime Platform check.

**Pull request merged**